synapse/logging/tracing.py

```diff
@@ -477,211 +457,196 @@ def whitelisted_homeserver(destination: str) -> bool:
+    start_time: Optional[int] = None,
+    record_exception: bool = True,
+    set_status_on_exception: bool = True,
+    end_on_exit: bool = True,
+    # For testing only
+    tracer: Optional["opentelemetry.trace.Tracer"] = None,
+) -> ContextManager["opentelemetry.trace.span.Span"]:
+    if opentelemetry is None:
         return contextlib.nullcontext()  # type: ignore[unreachable]
 
-    references = [opentracing.follows_from(context) for context in contexts]
-    scope = start_active_span(
-        operation_name,
-        child_of=child_of,
-        references=references,
+    # TODO: Why is this necessary to satisfy this error? It has a default?
+    #  ` error: Argument "kind" to "start_span" of "Tracer" has incompatible type "Optional[SpanKind]"; expected "SpanKind"  [arg-type]`
+    if kind is None:
+        kind = SpanKind.INTERNAL
+
+    span = start_span(
+        name=name,
+        context=context,
+        kind=kind,
+        attributes=attributes,
+        links=links,
         start_time=start_time,
-        tracer=tracer,
+        record_exception=record_exception,
+        set_status_on_exception=set_status_on_exception,
     )
 
-    if inherit_force_tracing and any(
-        is_context_forced_tracing(ctx) for ctx in contexts
-    ):
-        force_tracing(scope.span)
-
-    return scope
+    # Equivalent to `tracer.start_as_current_span`
+    return opentelemetry.trace.use_span(
+        span,
+        end_on_exit=end_on_exit,
+        record_exception=record_exception,
+        set_status_on_exception=set_status_on_exception,
+    )
 
 
 def start_active_span_from_edu(
-    edu_content: Dict[str, Any],
     operation_name: str,
-    references: Optional[List["opentracing.Reference"]] = None,
-    tags: Optional[Dict[str, str]] = None,
-    start_time: Optional[float] = None,
-    ignore_active_span: bool = False,
-    finish_on_close: bool = True,
-) -> "opentracing.Scope":
+    *,
+    edu_content: Dict[str, Any],
+) -> ContextManager["opentelemetry.trace.span.Span"]:
     """
     Extracts a span context from an edu and uses it to start a new active span
 
     Args:
+        operation_name: The label for the chunk of time used to process the given edu.
         edu_content: an edu_content with a `context` field whose value is
-        canonical json for a dict which contains opentracing information.
-
-        For the other args see opentracing.tracer
+            canonical json for a dict which contains opentracing information.
     """
-    references = references or []
-
-    if opentracing is None:
+    if opentelemetry is None:
         return contextlib.nullcontext()  # type: ignore[unreachable]
 
     carrier = json_decoder.decode(edu_content.get("context", "{}")).get(
-        "opentracing", {}
-    )
-    context = opentracing.tracer.extract(opentracing.Format.TEXT_MAP, carrier)
-    _references = [
-        opentracing.child_of(span_context_from_string(x))
-        for x in carrier.get("references", [])
-    ]
```

Comment:
I'm not sure if the OTEL `Context` include `Link`'s (what were `opentracing` references) now to make the equivalent code. Does it matter? (maybe not)